documentation/docs/30-advanced/40-service-workers.md

```diff
@@ -4,11 +4,76 @@ title: Service workers
+You can [disable automatic registration](/docs/configuration#serviceworker) if you need to register the service worker with your own logic or use another solution. The default registration looks something like this:
 
-Inside the service worker you have access to the [`$service-worker` module](/docs/modules#$service-worker). If your Vite config specifies `define`, this will be applied to service workers as well as your server/client builds.
+```js
+if ('serviceWorker' in navigator) {
+	addEventListener('load', function () {
+		navigator.serviceWorker.register('./path/to/service-worker.js');
+	});
+}
+```
+
+Inside the service worker you have access to the [`$service-worker` module](/docs/modules#$service-worker), which provides you with the paths to all static assets, build files and prerendered pages. You're also provided with an app version string which you can use for creating a unique cache name. If your Vite config specifies `define` (used for global variable replacements), this will be applied to service workers as well as your server/client builds.
+
+The following example caches the built app and any files in `static` eagerly, and caches all other requests as they happen. This would make each page work offline once visited.
+
+```js
+// @ts-nocheck Official TS Service Worker typings are still a work in progress.
+import { build, files, version } from '$service-worker';
+
+// Create a unique cache name for this deployment
+const CACHE_NAME = `cache-${version}`;
+
+self.addEventListener('install', (event) => {
+	// Create a new cache and add all files to it
+	async function addFilesToCache() {
+		const cache = await caches.open(CACHE_NAME);
+
+		await cache.addAll([
+			...build, // the app itself
+			...files  // everything in `static`
+		]);
+	}
+
+	event.waitUntil(addFilesToCache());
+});
+
+self.addEventListener('activate', (event) => {
+	// Remove previous cached data from disk
+	async function deleteOldCaches() {
+		const keyList = await caches.keys();
+		const cachesToDelete = keyList.filter((key) => key !== CACHE_NAME);
+		await Promise.all(cachesToDelete.map((key) => caches.delete(key)));
+	}
+
+	event.waitUntil(deleteOldCaches());
+});
+
+self.addEventListener('fetch', (event) => {
+	// Try to get the response from the cache, add to cache if not found
+	async function addToCache(request, response) {
+		const cache = await caches.open(CACHE_NAME);
+		await cache.put(request, response);
+	}
+
+	async function cacheFirst(request) {
+		const responseFromCache = await caches.match(request);
+		if (responseFromCache) {
+			return responseFromCache;
+		}
+		const response = await fetch(request);
+		addToCache(request, response.clone());
+		return response;
+	}
+
+	event.respondWith(cacheFirst(event.request));
+});
+```
+
+> Careful with caching too much for too long! Browsers have a limit on the amount they cache, and not everything should be cached, for example requests that contain dynamic data that change over time
```

Comment:
Since people are likely to just copy whatever we put in the docs, I'm not sure we should use a cache-first strategy here — it's a footgun. Cache-first makes sense for `build` and `files`, but for pages we should probably have a network-first strategy instead (and I think we should only fall back to the cache for GET requests?)

Reply:
https://github.com/sveltejs/kit/pull/7832

Reply:
Nuxt uses this workbox preset/tenplate: https://github.com/vite-pwa/vite-plugin-pwa/pull/390/files#diff-601038314cf235badafad57b9aca2c222c80d89c6a5153ee0708158a1efdb0d3R27